neural_compressor/compression/pruner/pruning.py

```diff
@@ -228,12 +228,13 @@ def _do_pruning(self):
                     pruner.fasterprune(layer_op_names)
                 for j in range(len(inputs)):
                     # the weights of current layer have been pruned, get the latest outputs as the inputs for next layer
-                    input_infos = self.gather_single_batch_from_dict(inp_dict, j)
-                    inputs[j] = layer(inputs[j], **input_infos)[0]
+                    other_infos = self.gather_single_batch_from_dict(other_input_infos, j)
```

Comment:
move the following no grad to funciton, too much space make the code a little ugly
        inputs, positional_inputs, other_input_infos = collect_layer_inputs(
            model=self._model, layers=layers, layer_idx=0, layer_inputs=self._dataloader, device=self.dev
        )

        **with torch.no_grad():**
            for i in tqdm(range(len(layers))):